lib/scheduler/scheduler.js

```diff
@@ -184,11 +184,17 @@ function connectEndpoint_p(endpoint, timeout, shouldContinue) {
             logger.trace("Refusing to delete /dev/null.");
             return;
           }
-          logger.trace('Normal exit, deleting log file ' + logFilePath);
-          fs.unlink(logFilePath, function(err) {
-            if (err)
-              logger.warn('Failed to delete log file ' + logFilePath + ': ' + err.message);
-          });
+
+          if (!appSpec || !appSpec.settings || !appSpec.settings.appDefaults || 
```

Comment:
What's the difference between `appSpec` and `self.$appSpec`? Are they different in practice?